### PR TITLE
Fixed #31365 -- Removed jQuery usage in SelectFilter2.js.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/31365

### CODE REVIEW

File: django/contrib/admin/static/admin/js/SelectFilter2.js [hunk 1]
```diff
@@ -236,11 +241,12 @@ Requires jQuery, core.js, and SelectBox.js.
     };
 
     window.addEventListener('load', function(e) {
-        $('select.selectfilter, select.selectfilterstacked').each(function() {
-            var $el = $(this),
-                data = $el.data();
-            SelectFilter.init($el.attr('id'), data.fieldName, parseInt(data.isStacked, 10));
-        });
+        var elements = document.querySelectorAll('select.selectfilter, select.selectfilterstacked');
+        for (var i = 0; i < elements.length; i++) {
```
Comment (jdufresne) [inline django/contrib/admin/static/admin/js/SelectFilter2.js]: `.forEach()` wasn't used here due to lack of support in the PhantomJS test runner.

**Comment by claudep** [comment]
Awesome! Do you think we could now remove the jQuery deps in `django.contrib.admin.widgets.FilteredSelectMultiple`?

**Comment by jdufresne** [comment]
Good catch! Thanks for pointing that out. It has been removed in the latest revision.

File: django/contrib/admin/static/admin/js/SelectFilter2.js (hunk 1 above)
Comment (jdufresne) [reply django/contrib/admin/static/admin/js/SelectFilter2.js]: Fixed by upgrading grunt-contrib-qunit to use puppeteer instead of PhantomJS.

**Comment by carltongibson** [comment]
buildbot, test on selenium

**Comment by carltongibson** [comment]
buildbot, test on selenium.

**Comment by carltongibson** [comment]
Hey @jdufresne. Any chance you can look at the Selenium failure here? https://djangoci.com/job/pull-requests-selenium/database=postgres,label=bionic-pr,python=python3.7/105/testReport/junit/admin_views.test_autocomplete_view/SeleniumTests/test_select_multiple/

Other than that, it looks good.

**Comment by jdufresne** [comment]
Yup, I'll investigate and report back when I have more information.

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by jdufresne** [comment]
I have rebased.

The Selenium test failure looks to be intermittent and inconsistent. As you can see, the latest Jenkins run passed. I can encounter the failure locally, but it is very rare and intermittent, so quite difficult to debug.

The error also appears to be preexisting. I'm able to trigger the same bug on the master branch, but again the occurrence is rare.

AFICT, the test is interacting with `select2.js` not `SelectFilter2.js`. So this may simply be a coincidence and unrelated to the change.

**Comment by jdufresne** [comment]
Here is an example of the test failing for a different PR:

**Comment by jdufresne** [comment]
Here is a PR that contains no changes: #12600 it shows the same test failing on master.

**Pull request merged**